plugins/gpio/gpioreaderthread.cpp

```diff
@@ -110,9 +110,15 @@ void ReadThread::run()
             uchar newVal = dataRead.toUInt();
             if (newVal != gpio->m_value)
             {
-                qDebug() << "Value read: GPIO:" << gpio->m_number << "val:" <<  newVal;
-                gpio->m_value = newVal;
-                emit valueChanged(gpio->m_number, gpio->m_value);
+                gpio->m_counter += 1;
+                if (gpio->m_counter > 3) {
```

Comment:
You said you don't want to change the code style, but still your style places parenthesis at the end of the line.